TODO.md

```diff
@@ -0,0 +1,6 @@
+# stuff that I will definitely* do in my ample free time
+
+* Figure out how to have poetry insist that main works for python 3.7 while also supporting flake8 and pytest
```

Comment:
As above, the easiest solution is probably to drop this requirement and just accept python >=3.8 once we've moved past Ansible, but this is another space where Docker could be really helpful as we could run our CI checks in both a base image built from Python 3.7 and one from Python 3.8 if we really wanted to (but generally this just means we're supporting twice as many issues, and not worth the cost!)